core/src/main/scala/com/salesforce/op/stages/impl/tuning/DataCutter.scala

```diff
@@ -104,11 +128,35 @@ class DataCutter(uid: String = UID[DataCutter]) extends Splitter(uid = uid) with
    */
   override def validationPrepare(data: Dataset[Row]): Dataset[Row] = {
     val dataPrep = super.validationPrepare(data)
+    val labelMetaArr = getLabelsFromMetadata(dataPrep)
+    val labelSet = getLabelsToKeep.toSet
 
-    val keep: Set[Double] = getLabelsToKeep.toSet
-    val dataUse = dataPrep.filter(r => keep.contains(r.getDouble(0)))
+    // discount unseen label
+    val integralLabelSet = 0.until(labelMetaArr.size - 1)
+      .map(_.toDouble)
+      .toSet
 
-    dataUse
+    if (integralLabelSet == labelSet) {
+      log.info("Label sets identical in dataset metadata and datacutter, skipping label trim")
+      dataPrep
+    } else {
+      log.info(s"Dropping rows with columns not in $labelSet")
+      val labelColName = dataPrep.columns(0)
+
+      val na = NominalAttribute.defaultAttr.withName(labelColName)
+      val metadataNA = if (labelMetaArr.isEmpty) {
+        na.withNumValues(labelSet.size)
+      } else {
+        // trim the metadata
```

Comment:
what does `trim the metadata` and `trim the dataframe` means? please add more explanations  ;)